mantle/cmd/ore/aws/delete-image.go

```diff
@@ -57,7 +57,7 @@ func runDeleteImage(cmd *cobra.Command, args []string) {
 	}
 
 	if snapshotID != "" {
-		err := API.RemoveBySnapshotTag(snapshotID, allowMissing)
+		err := API.RemoveBySnapshotTag(snapshotID, amiID, allowMissing)
```

Comment:
I think rather than changing this function we should change the input `--snapshot` to accept a new special value for snapshot ID of `detectFromAMI`.

If this value was passed in then we would:

- First determine if AMI exists
- If AMI exists determine snapshot ID from it and set that value

then we can continue with the rest of the code

Reply:
we would then change the calling GC code to pass in `--snapshot detectFromAMI` if no snapshot was set in the json.